packages/commonwealth/server/util/tokenBalanceCache/tokenBalanceCache.ts

```diff
@@ -118,31 +142,120 @@ export class TokenBalanceCache {
       },
     });
 
+    if (!chainNode) {
+      const msg = `ChainNode with eth_chain_id ${options.sourceOptions.evmChainId} does not exist`;
+      log.error(msg);
+      rollbar.error(msg);
+      return {};
+    }
+
+    let freshBalances: Balances = {};
     switch (options.balanceSourceType) {
       case BalanceSourceType.ETHNative:
-        return await __getEthBalances.call(this, {
+        freshBalances = await __getEthBalances.call(this, {
           chainNode,
-          addresses: validatedAddress,
+          addresses: validatedAddresses,
         });
+        break;
       case BalanceSourceType.ERC20:
-        return await __getErc20Balances.call(this, {
+        freshBalances = await __getErc20Balances.call(this, {
           chainNode,
-          addresses: validatedAddress,
+          addresses: validatedAddresses,
           contractAddress: options.sourceOptions.contractAddress,
         });
+        break;
       case BalanceSourceType.ERC721:
-        return await __getErc721Balances.call(this, {
+        freshBalances = await __getErc721Balances.call(this, {
           chainNode,
-          addresses: validatedAddress,
+          addresses: validatedAddresses,
           contractAddress: options.sourceOptions.contractAddress,
         });
+        break;
       case BalanceSourceType.ERC1155:
-        return await __getErc1155Balances.call(this, {
+        freshBalances = await __getErc1155Balances.call(this, {
           chainNode,
-          addresses: validatedAddress,
+          addresses: validatedAddresses,
           contractAddress: options.sourceOptions.contractAddress,
           tokenId: options.sourceOptions.tokenId,
         });
+        break;
     }
+
+    await this.cacheBalances(options, freshBalances);
+
+    return { ...freshBalances, ...cachedBalances };
```

Comment:
Shouldn't the fresh balances be spread *after* the cached balances?

Reply:
It's fine if there's no overlap in keys between the fresh and cached objects. If there is overlap, then the cached values will override the fresh.